src/python/pants/rules/core/cloc.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import Path
+from typing import Set
+
+from pants.base.build_environment import get_buildroot
+from pants.base.specs import Specs
+from pants.engine.console import Console
+from pants.engine.fs import (
+  Digest,
+  DirectoriesToMerge,
+  FileContent,
+  FilesContent,
+  InputFilesContent,
+  Snapshot,
+  UrlToFetch,
+)
+from pants.engine.goal import Goal
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.legacy.graph import HydratedTargets, TransitiveHydratedTargets
+from pants.engine.rules import console_rule, rule
+from pants.engine.selectors import Get
+from pants.util.objects import datatype
+
+
+class DownloadedClocScript(datatype([('script_path', str), ('digest', Digest)])):
+  """Cloc script as downloaded from the pantsbuild binaries repo."""
+
+
+#TODO(#8459) - It would be nice if we weren't hardcoding a specific version of the cloc binary here.
+@rule
+def download_cloc_script() -> DownloadedClocScript:
+  url = "https://binaries.pantsbuild.org/bin/cloc/1.80/cloc"
```

Comment:
Oh, hm. The URL actually needs to be constructed using `BinaryUtil.Factory`, because although it's fine to hardcode the version, we can't really hardcode the URL (users use these tools from behind proxies).

Reply:
@stuhood since this is not trivial to do (and since we already have an open ticket https://github.com/pantsbuild/pants/issues/7790 for this), in the interest of getting this PR merged I've removed the code that gets rid of the v1 version of `cloc` and renamed the v2 version `fast-cloc`. Since we're no longer dropping support for the v1 `cloc`, I think this should be okay to merge, and then fix the hardcoding in a subsequent commit.

Reply:
Yep.